TowerDefense/GameObjects/Node.cs

```diff
@@ -40,12 +40,26 @@ public Node(Point actualPos, Point simplePos, Texture2D tex) : base(tex, actualP
         {
             this.actualPos = actualPos;
             this.simplePos = simplePos;
+
+            spriteHeight = 32;
+            spriteWidth = 32;
+            currentFrame = 0;
+        }
+
+        public override bool Update(GameTime gameTime)
+        {
+            frameCount = portal ? 4 : 0;
+            currentFrame = portal ? currentFrame : 0;
+            return base.Update(gameTime);
         }
+
         public override void Draw(SpriteBatch batch) 
         {
             base.Draw(batch);
-            if ( tex2 != null )
-                batch.Draw(tex2, actualPos.ToVector2(), null, Color);
+            if (tex2 != null)
+            {
+                batch.Draw(tex2, Position.ToVector2(), new Rectangle(new Point(currentFrame * spriteWidth, 0), new Point(spriteWidth, spriteHeight)), Color, 0, Vector2.Zero, 1, SpriteEffects.None, 0);
```

Comment:
Nodes are the squares you can put stuff.  If you place a wall on it, you see the grass texture still, and on top of that it draws a wall.  It's most noticeable with the portal since it's very see through, without 2 textures, you would replace the grass with a portal on smokewhite or whatever the background clears too.